### PR TITLE
Upgrade for ckan 2.9

### PR DESCRIPTION
**Pull request by markstuart**
Following guidance here: https://github.com/ckan/ckan/wiki/Python-3-migration-guide-for-extensions we have updated this extension to be ckan 2.9 and 2.8 compatible.

### CODE REVIEW

File: ckanext/dia/harvester/dcat.py [hunk 1]
```diff
@@ -9,50 +11,28 @@
...
-    non_iso_8601_to_iso_8601_frequency = {
-        "6-Monthly": "Semiannual",
-        "Annually": "Annual",
-        "As Required": "Irregular",
-        "Other-Unknown": "Irregular",
-        "Realtime": "Continuously updated",
-        "Static": "Irregular",
-        "Unknown": "Irregular",
-        "Yearly": "Annual",
-    }
```
Comment (markstuart) [inline ckanext/dia/harvester/dcat.py:54]: These objects weren't used or referenced anywhere.

File: ckanext/dia/schema.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
-from ckan.lib.navl.validators import ignore_missing, not_empty, ignore, not_missing
+from ckan.lib.navl.validators import (
+    ignore_missing, not_empty, ignore, not_missing)
 
 
 def default_extras_schema():
     return {
         'id': [ignore],
-        'key': [not_empty, unicode],
+        'key': [not_empty, str],
```
Comment (markstuart) [inline ckanext/dia/schema.py]: Hmm, not sure about this one

Comment (markstuart) [reply ckanext/dia/schema.py]: maybe should be `six.text_type`??

File: requirements-py2.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# To regenerate from the project's Pipfile, run:
+#
+#    pipenv lock --requirements
+#
+
+-i https://pypi.org/simple
+aenum==3.1.0; python_version < '3.6'
+future==0.18.2
+pycountry==18.12.8
+pyproj==2.2.2
```
Comment (markstuart) [inline requirements-py2.txt:12]: The toplevel requirements file for the data.govt.nz core repo had recorded the previously installed/frozen versions

File: ckanext/dia/schema.py (hunk 2 above)
Comment (niallmcandrew-work) [reply ckanext/dia/schema.py]: Yes I think this should be. Did `futurize` not pick it up?

Comment (markstuart) [reply ckanext/dia/schema.py]: Nope, will sort this now

**Pull request merged**